### PR TITLE
[WIP] CI run for minimal-partr PR (3/3)

### PR DESCRIPTION
**Pull request by vtjnash**

### CODE REVIEW

**Comment by KristofferC** [comment]
Windows precompile generator strikes again... Sorry :(

File: base/task.jl [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## basic task functions and TLS
 
+const ThreadSyncronizer = GenericCondition{Threads.SpinLock}
```
Comment (vchuravy) [inline base/task.jl]: `ThreadSynchronizer`

File: base/task.jl [hunk 2]
```diff
@@ -354,3 +372,261 @@ function timedwait(testcb::Function, secs::Float64; pollint::Float64=0.1)
...
+julia> yield();
+
+julia> istaskstarted(b)
+true
+
+julia> istaskdone(b)
+true
+```
+"""
+function schedule(t::Task, @nospecialize(arg); error=false)
```
Comment (vchuravy) [inline base/task.jl]:
```suggestion
function schedule(t::Task, @nospecialize(arg)=nothing; error=false)
```

File: src/partr.c [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+/* a task heap */
+typedef struct taskheap_tag {
+    jl_mutex_t lock;
+    jl_task_t **tasks;
+    int16_t ntasks, prio;
+} taskheap_t;
+
+/* multiqueue parameters */
+static const int16_t heap_d = 8;
+static const int heap_c = 4;
```
Comment (vchuravy) [inline src/partr.c]: Is this the same as `MULTIQ_HEAP_C `?

File: base/task.jl (hunk 2 above)
Comment (vtjnash) [reply base/task.jl]: https://github.com/JuliaLang/julia/pull/30731/files/a71cbcfa4feb8cce8efbb367d1d78045126e85e8#diff-121162110854f09c3be5b39209646009R455

File: src/partr.c (hunk 3 above)
Comment (kpamnany) [reply src/partr.c]: It is. Should I move to using `#define`s like `MULTIQ_HEAP_C` or leave this `const`?

**Comment by StefanKarpinski** [comment]
Superseded by https://github.com/JuliaLang/julia/pull/31398.

**Pull request closed without merging**